alibi/explainers/backends/cfrl_tabular.py

```diff
@@ -363,55 +371,55 @@ def sample_categorical(X_hat_cat_split: List[np.ndarray],
+        values for a feature.
     stats
         Dictionary of statistic of the training data. Contains the minimum and maximum value of each numerical
         feature in the training set. Each key is an index of the column and each value is another dictionary
         containing `min` and `max` keys.
-    category_map
-        Dictionary of category mapping. The keys are column indexes and the values are lists containing the possible
-        values for a feature.
 
     Returns
     -------
-    Most probable sample according to the auto-encoder reconstruction, sampled according to the conditional vector
-    and the dictionary of statistics. This method assumes that the input array, `x_ohe`, is encoded as follows:
-    first columns correspond to the numerical features, and the rest are one-hot encodings of the categorical
-    columns.
+    X_ohe_hat_split
+        Most probable reconstruction sample the autoencoder, sampled according to the conditional vector and the
+        dictionary of statistics. This method assumes that the input array, `X_ohe` , has the first columns
+        corresponding to the numerical features, and the rest are one-hot encodings of the categorical columns.
     """
     X_ohe_num_split, X_ohe_cat_split = split_ohe(X_ohe, category_map)
-    C_num_split, C_cat_split = split_ohe(C, category_map)
+    C_num_split, C_cat_split = split_ohe(C, category_map) if (C is not None) else (None, None)
 
-    sampled_num, sampled_cat = [], []  # list of sampled numerical columns and sampled categorical columns
+    X_ohe_hat_split = []  # list of sampled numerical columns and sampled categorical columns
     num_feat, cat_feat = len(X_ohe_num_split), len(X_ohe_cat_split)
 
     if num_feat > 0:
-        # sample numerical columns
-        sampled_num = sample_numerical(X_hat_num_split=X_hat_split[:num_feat],
-                                       X_ohe_num_split=X_ohe_num_split,
-                                       C_num_split=C_num_split,
-                                       stats=stats)
+        # Sample numerical columns
+        X_ohe_hat_split += sample_numerical(X_hat_num_split=X_hat_split[:num_feat],
+                                            X_ohe_num_split=X_ohe_num_split,
+                                            C_num_split=C_num_split,
+                                            stats=stats)
 
     if cat_feat > 0:
-        # sample categorical columns
-        sampled_cat = sample_categorical(X_hat_cat_split=X_hat_split[-cat_feat:],
-                                         C_cat_split=C_cat_split)
+        # Sample categorical columns
+        X_ohe_hat_split += sample_categorical(X_hat_cat_split=X_hat_split[-cat_feat:],
+                                              C_cat_split=C_cat_split)
 
-    return sampled_num + sampled_cat
+    return X_ohe_hat_split
 
 
 def get_he_preprocessor(X: np.ndarray,
```

Comment:
Just noticed that this function has a mutable default `dict()`, see https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html